Meta.Rmd

```diff
@@ -8,37 +8,37 @@ source("common.R")
 \index{metaprogramming}
 \index{non-standard evaluation}
 
-One of the most intriguing things about R is its capability for __metaprogramming__: the idea that code is itself data, and can be inspected and modified programmatically. This is a powerful idea and deeply influences much R code. At a simple level it allows you to write `library(purrr)` instead of `library("purrr")` and enables `plot(x, sin(x))` to label the axes with `x` and `sin(x)`. At a deeper level it allows `y ~ x1 + x2` to represent a model that predicts the value of `y` from `x1` and `x2`, `subset(df, x == y)` to be translated to `df[df$x == df$y, , drop = FALSE]`, and `dplyr::filter(db, is.na(x))` to generate the SQL `WHERE x IS NULL` when `db` is a remote database table.
+One of the most intriguing things about R is its ability to do __metaprogramming__. This is the idea that code is data, and can be inspected and modified programmatically. This is a powerful idea; one that deeply influences much R code. At the most basic level, it allows you to do things like write `library(purrr)` instead of `library("purrr")` and enable `plot(x, sin(x))` to automatically label the axes with `x` and `sin(x)`. At a deeper level, it allows you to do things like use `y ~ x1 + x2` to represent a model that predicts the value of `y` from `x1` and `x2`, to translate `subset(df, x == y)` into `df[df$x == df$y, , drop = FALSE]`, and to use `dplyr::filter(db, is.na(x))` to generate the SQL `WHERE x IS NULL` when `db` is a remote database table.
 
-Closely related to metaprogramming is __non-standard evaluation__, or NSE for short. This term is commonly used to describe the behaviour of R functions, but there are two problems with it. Firstly, NSE is actually a property of an argument (or arguments) of a function, so talking about NSE functions is a little sloppy. Secondly, it's confusing to define something by what it is not (standard), so in this book I'll teach you more precise vocabulary.
+Closely related to metaprogramming is __non-standard evaluation__, NSE for short. This term is commonly used to describe the behaviour of R functions, but there are two problems with this. Firstly, NSE is actually a property of a function's argument (or arguments), so talking about NSE functions is a little sloppy. Secondly, it's confusing to define something by what it's not (standard), so in this book I'll teach you a more precise vocabulary.
```

Comment:
it implies the term; this implies the behaviour — and it's the term I have problems with.

Also I like argument coming first because it emphasises the distinction from function.

Reply:
"Closely related to metaprogramming is __non-standard evaluation__, NSE for short. This term, which is commonly used to describe the behaviour of R functions, is problematic in my opinion. Firstly, NSE is actually a property of the argument (or arguments) of a function, so talking about NSE functions is a little sloppy. Secondly, it's confusing to define something by what it's not (standard), so in this book I'll introduce more precise vocabulary."

Reply:
Perfect (although I'd remove "in my opinion")